tests/test_utils/test_timing_utils.py

```diff
@@ -0,0 +1,167 @@
+"""Tests for timing utility functions."""
+
+import unittest
+import tempfile
+from datetime import datetime
+from pathlib import Path
+from zoneinfo import ZoneInfo
+import pandas as pd
+from unittest.mock import patch
+
+from aind_metadata_mapper.utils.timing_utils import (
+    convert_ms_since_midnight_to_datetime,
+    find_latest_timestamp_in_csv_files,
+    _read_csv_safely,
+    _extract_max_timestamp,
+)
+
+
+class TestTimingUtils(unittest.TestCase):
+    """Test timing utility functions."""
+
+    def test_convert_ms_since_midnight_to_datetime_default_timezone(self):
+        """Test conversion with default timezone."""
+        base_date = datetime(2024, 1, 1, tzinfo=ZoneInfo("UTC"))
+
+        # Test with no timezone specified (uses system default)
+        result = convert_ms_since_midnight_to_datetime(0.0, base_date)
+        self.assertIsNotNone(result.tzinfo)
+        self.assertEqual(result.hour, 0)
+        self.assertEqual(result.minute, 0)
+
+    def test_read_csv_safely_error_conditions(self):
+        """Test CSV reading with error conditions."""
+        with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
I tend to discourage the use of writing to the user's file system during unit tests. For reading small files, we've been putting pre-made files into the test resources directory. For writes, we usually mock the write operation.